src/API/Google/MerchantReport.php

```diff
@@ -52,6 +56,87 @@ public function __construct( ShoppingContent $service, ProductHelper $product_he
 		$this->product_helper = $product_helper;
 	}
 
+	/**
+	 * Get ProductView Query response.
+	 *
+	 * @param string|null $next_page_token The next page token.
+	 * @return array Associative array with product statuses and the next page token.
+	 *
+	 * @throws Exception If the product view report data can't be retrieved.
+	 */
+	public function get_product_view_report( $next_page_token = null ): array {
+		$batch_size = apply_filters( 'woocommerce_gla_product_view_report_page_size', 1000 );
+
+		try {
+			$product_view_data = [
+				'statuses'  => [],
+				'next_page' => null,
+			];
+
+			$query = new MerchantProductViewReportQuery(
+				[
+					'next_page' => $next_page_token,
+					'per_page'  => $batch_size,
+				]
+			);
+
+			$response = $query
+			->set_client( $this->service, $this->options->get_merchant_id() )
+			->get_results();
+
+			foreach ( $response->getResults() as $row ) {
+
+				/** @var ProductView $product_view  */
+				$product_view = $row->getProductView();
+
+				$wc_product_id     = $this->product_helper->get_wc_product_id( $product_view->getId() );
+				$mc_product_status = $this->convert_aggregated_status_to_mc_status( $product_view->getAggregatedDestinationStatus() );
+
+				// Skip if the product id does not exist
```

Comment:
Not a big deal, but technically we've only tried extracting the product ID from the string, we haven't actually checked if it exists.

This makes me wonder though, we are extracting from `$product_view->getId()`, why is both the `offer_id` and `id` included in the query MerchantProductViewReportQuery then? Not a big deal to leave it there, but was just curious. Seems like `getOfferId()` would return a much shorter string.